x-pack/plugins/files/server/routes/find.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { schema, TypeOf } from '@kbn/config-schema';
+import type { Ensure } from '@kbn/utility-types';
+import type { FilesRouter } from './types';
+
+import { FindFilesHttpEndpoint, FILES_API_ROUTES } from '../../common/api_routes';
+import type { FilesRequestHandler } from './types';
+
+export const method = 'post' as const;
+
+const stringOrArrayOfStrings = schema.oneOf([schema.string(), schema.arrayOf(schema.string())]);
```

Comment:
If it is possible to limit array length, would be great to limit it to something, say 10. 

---

And also would be great to set limit on strings, say 128 chars. Or maybe more granular, say: 32 chars for everything and 256 for `name`.

Reply:
Yeah, this sounds great. I love the idea of handling cases like this.